Lib/ntpath.py

```diff
@@ -659,6 +658,8 @@ def realpath(path):
             path = _getfinalpathname(path)
             initial_winerror = 0
         except OSError as ex:
+            if strict:
```

Comment:
It may as well handle `ERROR_CANT_RESOLVE_FILENAME` here instead of calling `_getfinalpathname` again.

```python
    # ERROR_CANT_RESOLVE_FILENAME (1921) is from exceeding the
    # max allowed number of reparse attempts (currently 63), which
    # is either due to a loop or a chain of links that's too long.
    if strict or ex.winerror == 1921:
        raise
```
I agree that there's no point in returning a path that has a link that can't be resolved due to a cycle or too many links. It's a bad path, and someone needs to be made aware of it by raising an exception. But I haven't read the reasons and use cases that led to the current design.